craft_application/application.py

```diff
@@ -223,12 +227,12 @@ def _resolve_project_path(self, project_dir: pathlib.Path | None) -> pathlib.Pat
          in multiple places within the project directory.
         """
         if project_dir is None:
-            project_dir = pathlib.Path.cwd()
+            project_dir = self.project_dir
         return (project_dir / f"{self.app.name}.yaml").resolve(strict=True)
 
     def get_project(
         self,
-        project_dir: pathlib.Path | None = None,
+        project_dir: None = None,
```

Comment:
my personal feeling here is that we should just remove this parameter altogether. My reasoning is that I think the buildplanner/grammar stuff will already introduce breaking changes, so if we're doing a 2.0 we might as well clean house while we're at it. And the fixes downstream should be straightforward.

Reply:
I'm more than happy to do so! I wasn't sure whether this would come in with a 2.0 or after, but if we're targeting 2.0 now that makes my life way easier.